ShadowDOM/explainer.md

```diff
@@ -0,0 +1,401 @@
+
+```html
+<style> 
+  @sheet sheet1 { *: background-color: gray; } 
+  @sheet sheet2 { *: color: blue; } 
+</style> 
+
+<template shadowrootmode="open" adoptedStylesheets="sheet1, sheet2"> 
+  <span>I'm in the shadow DOM</span> 
+</template> 
+ ```
+
+In this example, developers could define styles in a `<style>` block using an `@sheet` rule to create named style sheets. The `adoptedStylesheets` property allows Shadow DOMs to specify which stylesheets they want to adopt without impacting the main document, improving ergonomics.  
+
+The JavaScript version of this could also support CSS modules: 
+```js
+@sheet sheet1 { 
+  :host { 
+    display: block; 
+    background: red; 
+  } 
+} 
+ 
+@sheet sheet2 { 
+  p { 
+    color: blue; 
+  } 
+} 
+
+<script> 
+import {sheet1, sheet2} from './styles1and2.css' assert {type: 'css'}; 
+... 
+shadow.adoptedStyleSheets = [sheet1, sheet2]; 
+</script>
+```
+This approach could be combined with other approaches listed in this document. Note that `@sheet` is not implemented by any rendering engine as of September 2024. 
+
+### [Id-based `adoptedstylesheet` attribute on template](https://github.com/WICG/webcomponents/issues/939#issue-971914425) 
+This proposal will add a new markup-based `adoptedstylesheets` property that closely matches the existing JavaScript property.  The behavior would be just like the `adoptedStyleSheet` property that already exists in JavaScript, except it would accept a list of id attributes instead of a `ConstructableStylesheet` JavaScript object. 
+```html
+<style type="css" id="shared_shadow_styles"> 
+    :host {  
+      color: red  
+    }  
+</style>
+```
+
+or 
+
+```html
+<link rel=”stylesheet” href=”styles.css” id=”external_shared_shadow_styles”>
+```
+Web authors can use the `adoptedstylesheets` property on the `<template>` element to associate the stylesheets with a declarative shadow root. 
+```html
+<template shadowrootmode="open" adoptedstylesheets="shared_shadow_styles, external_shared_shadow_styles">  
+      <!-- -->  
+</template>
+```
+One requirement of this approach is that the current `adoptedStylesheets` JavaScript property would need to lift the “constructable” requirement for `adoptedStylesheets`. This was recently agreed upon by the CSSWG but has not been implemented yet: [ Can we lift the restriction on constructed flag for adoptedStylesheets?](https://github.com/w3c/csswg-drafts/issues/10013#issuecomment-2165396092)
+
+One limitation of this approach is that shared styles that need to be applied exclusively to shadow roots (and not the main document) will need to include a CSS `:host` selector. This is not necessary for JavaScript-based adopedStylesheets but will be necessary for declarative stylesheets, as there is currently no way in HTML to create stylesheets without applying them to the document they are defined in. This could also be addressed via a new type value on `<style` tags and rel value on `<link>` tags, potentially `“adopted-css”`. 
```

Comment:
```suggestion
One limitation of this approach is that shared styles that need to be applied exclusively to shadow roots (and not the main document) will need to include a CSS `:host` selector. This is not necessary for JavaScript-based adopedStylesheets but will be necessary for declarative stylesheets, as there is currently no way in HTML to create stylesheets without applying them to the document they are defined in. This could also be addressed via a new type value on `<style>` tags and rel value on `<link>` tags, potentially `“adopted-css”`. 
```